File: JavaScript/0242-有效的字母异位词.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
function isAnagram(s, t) {
  if (s.length !== t.length) {
    return false;
  }
  const record = new Array(26).fill(0),
    n = s.length,
    base = 'a'.charCodeAt();

  for (let i = 0; i < n; i++) {
    record[s[i].charCodeAt() - base]++;
  }

  for (let i = 0; i < n; i++) {
    if (--record[t[i].charCodeAt() - base] < 0) {
      return false;
    }
  }
  return true;
}

File: Java/0242-有效的字母异位词.java
class Solution {
  public boolean isAnagram(String s, String t) {
    if (s.length() != t.length()) return false;
    int[] record = new int[26];
    int n = s.length();
    for (int i = 0; i < n; i++) record[s.charAt(i) - 'a']++;
    for (int i = 0; i < n; i++) if (--record[t.charAt(i) - 'a'] < 0) return false;
    return true;
  }
}
